include/sharpSAT/primitive_types.h

```diff
@@ -8,20 +8,189 @@
 #ifndef SHARP_SAT_PRIMITIVE_TYPES_H_
 #define SHARP_SAT_PRIMITIVE_TYPES_H_
 
```

Comment:
This goes for all the newly introduces types: we should try to avoid using (and thus defining) the postfix versions of increment and decrement. I am 99% sure that they aren't really needed, and when the binary is compiled with full optimizations, they do not introduce performance penalty, but when working on a debug build, they are needlessly costly.

Reply:
Done.